torch_xla/distributed/fsdp/xla_fully_sharded_data_parallel.py

```diff
@@ -351,6 +421,7 @@ def clip_grad_norm_(
       self,
```

Comment:
Sorry I didn't notice `clip_grad_norm_` in the last PR. When is this method called?

Reply:
This method is called to apply gradient clipping similar to [torch.nn.utils.clip_grad_norm_](https://pytorch.org/docs/stable/generated/torch.nn.utils.clip_grad_norm_.html), but based on the full gradient norm (rather than the norm of the sharded gradients in each worker as in FSDP)

Reply:
Sorry, I meant I didn't see it been called within FSDP. Do you mean it's supposed to be called outside of FSDP?

Reply:
Yes, it should be called outside of FSDP (it's not part of the model, but rather a part of the optimization strategy) and before the gradients are consumed by the optimizer, similar to calling torch.nn.utils.clip_grad_norm_ outside the model. An example usage is in https://github.com/ronghanghu/vit_10b_fsdp_example/blob/master/run_vit_training.py#L264-L273 (changing `torch.nn.utils.clip_grad_norm_` to `model.clip_grad_norm_` in the training loop for the FSDP case)